### DISCUSSION TITLE
UI.Handle coordinate system

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#350] [Q&A] [answered]
Just to confirm: does `UI.Handle()` always store the updated pose in the current local coordinate system, and so not in untransformed world-space? Testing with the code below it seems that when I put the two cubes in the same world-space (visual) location pose2 is different by -1 in X, suggesting the handle interaction indeed stores the pose relative to the current transform on the hierarchy.

```
                UI.Handle("Cube 1", ref pose1, cube.Bounds);
                cube.Draw(pose1.ToMatrix());

                Hierarchy.Push(Matrix.T(1f, 0, 0));

                    UI.Handle("Cube 2", ref pose2, cube.Bounds);
                    cube.Draw(pose2.ToMatrix());

                Hierarchy.Pop();
```

**Comment by maluoi** [comment, marked as answer]
That is correct! Handle poses are indeed Hierarchy space relative.
